test-utils/state-viewer/src/apply_chain_range.rs

```diff
@@ -0,0 +1,178 @@
+                    shard_id,
+                    block_hash,
+                    prev_block.chunks()[shard_id as usize].height_included(),
+                )
+                .unwrap();
+            let receipts = collect_receipts_from_response(&receipt_proof_response);
+
+            let chunk_inner = chunk.cloned_header().take_inner();
+            let is_first_block_with_chunk_of_version =
+                check_if_block_is_first_with_chunk_of_version(
+                    &mut chain_store,
+                    runtime_adapter.as_ref(),
+                    block.header().prev_hash(),
+                    shard_id,
+                )
+                    .unwrap();
+            runtime_adapter
+                .apply_transactions(
+                    shard_id,
+                    chunk_inner.prev_state_root(),
+                    None,
+                    height,
+                    block.header().raw_timestamp(),
+                    block.header().prev_hash(),
+                    block.hash(),
+                    &receipts,
+                    chunk.transactions(),
+                    chunk_inner.validator_proposals(),
+                    prev_block.header().gas_price(),
+                    chunk_inner.gas_limit(),
+                    &block.header().challenges_result(),
+                    *block.header().random_value(),
+                    true,
+                    is_first_block_with_chunk_of_version,
+                    None,
+                )
+                .unwrap()
+        } else {
+            let chunk_extra = chain_store
+                .get_chunk_extra(block.header().prev_hash(), &shard_uid)
+                .unwrap()
+                .clone();
+
+            runtime_adapter
+                .apply_transactions(
+                    shard_id,
+                    chunk_extra.state_root(),
+                    None,
+                    block.header().height(),
+                    block.header().raw_timestamp(),
+                    block.header().prev_hash(),
+                    &block.hash(),
+                    &[],
+                    &[],
+                    chunk_extra.validator_proposals(),
+                    block.header().gas_price(),
+                    chunk_extra.gas_limit(),
+                    &block.header().challenges_result(),
+                    *block.header().random_value(),
+                    false,
+                    false,
+                    None,
+                )
+                .unwrap()
+        };
+
+        let (outcome_root, _) =
+            ApplyTransactionResult::compute_outcomes_proof(&apply_result.outcomes);
+        let chunk_extra = ChunkExtra::new(
+            &apply_result.new_root,
+            outcome_root,
+            apply_result.validator_proposals,
+            apply_result.total_gas_burnt,
+            near_config.genesis.config.gas_limit,
+            apply_result.total_balance_burnt,
+        );
```

Comment:
This is where the problem is. When the block contains no new chunk, we copy the previous chunk extra and update state_root

Reply:
Fixed, thanks!